tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/types/decimaltype/BDecimalBFloatComparisonTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.test.types.decimaltype;
+
+import org.ballerinalang.launcher.util.BCompileUtil;
+import org.ballerinalang.launcher.util.BRunUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.model.values.BBoolean;
+import org.ballerinalang.model.values.BDecimal;
+import org.ballerinalang.model.values.BFloat;
+import org.ballerinalang.model.values.BValue;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.ballerinalang.compiler.semantics.model.types.util.Decimal;
+
+/**
+ * This class is used to test the behaviours of BDecimal and BFloat types by comparing them.
+ * This class contains tests,
+ * 1) demonstrating how very large numbers can be represented using BDecimal type.
+ * 2) demonstrating how BDecimal type solves the floating point error problem that exists in the BFloat type.
+ */
```

Comment:
Since this a newly added file shall we add the version?

Reply:
Added with commit 3957167